### PR TITLE
Pair distribution function

### PR DESCRIPTION
**Pull request by ThoChat**
to solve issue #325

### CODE REVIEW

File: pedpy/_version.py
Comment (chraibi) [inline pedpy/_version.py:1]: is this file necessary?

File: pedpy/methods/other_characterisation_methods.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    # Compute the number of possible combinaison of id per frame
+    N_val = 0
+    for _, df_f in gdf:
+        ids = len(df_f[ID_COL].unique())  # Number of unique ids
+        N_val += int(ids * (ids - 1) / 2)  # Calculate combinations of ids
+
+    dist_array = np.empty((N_val))  # Initialize array to store results
+    index = 0 
+    for _, df_f in gdf:  # Iterate over groups (frames)
+        gdf_f_id = df_f.groupby(ID_COL)
```
Comment (chraibi) [inline pedpy/methods/other_characterisation_methods.py:61]: A triple for-loop and twice groupifying?

File: pedpy/methods/pair_distribution_function_calculator.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+from pedpy.column_identifier import  FRAME_COL, ID_COL,X_COL,Y_COL,V_X_COL,V_Y_COL
+from pedpy.data.geometry import MeasurementArea
+from pedpy.data.trajectory_data import TrajectoryData
+from pedpy.methods.speed_calculator import (
+    SpeedCalculation,
+    compute_individual_speed)
+
+
+def compute_pair_distibution_function(
```
Comment (chraibi) [inline pedpy/methods/pair_distribution_function_calculator.py:16]: is this the same like the other file "other_methods..."

**Review by chraibi — CHANGES_REQUESTED** [review]
I think you also need to use the formating script before you commit/push

**Comment by chraibi** [comment]
In general, I'm confused `compute_pair_distibution_function` being in two different files.

File: pedpy/_version.py
Comment (schroedtert) [inline pedpy/_version.py:1]: This file should not be added to the repo

File: pedpy/methods/pair_distribution_function.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    # Suppress warnings
+    warnings.filterwarnings("ignore")
+
+    try:
+        with np.errstate(divide="raise"):
+            pair_distribution = pd_bins_normalised / pd_ni_bins_normalised
+        warnings.filterwarnings("default")  # reset wrnings values
+
+    except FloatingPointError:
+        warnings.filterwarnings("default")  # reset wrnings values
```
Comment (chraibi) [inline pedpy/methods/pair_distribution_function.py:91]: warning-values

**Review by chraibi — APPROVED** [review]

**Review by chraibi — APPROVED** [review]

**Review by chraibi — APPROVED** [review]

**Review by chraibi — DISMISSED** [review]

**Review by schroedtert — DISMISSED** [review]
The new functions are not shown in the API guide on the website

**Review by chraibi — APPROVED** [review]

**Pull request merged**